package/piwebcam/multi-gadget.sh

```diff
@@ -64,17 +66,13 @@ EOF
 config_usb_webcam () {
   mkdir -p functions/uvc.usb0/control/header/h
 
-  config_frame mjpeg m  640  360
-  config_frame mjpeg m  640  480
-  config_frame mjpeg m  800  600
-  config_frame mjpeg m 1024  768
-  config_frame mjpeg m 1280  720
-  config_frame mjpeg m 1280  960
-  config_frame mjpeg m 1440 1080
-  config_frame mjpeg m 1536  864
-  config_frame mjpeg m 1600  900
-  config_frame mjpeg m 1600 1200
-  config_frame mjpeg m 1920 1080
+  grep -E "(mjpeg|raw|mp4)\s+[[:digit:]]+\s+[[:digit:]]+" $VIDEO_FORMATS_FILE | sed -E "s/\s+/ /g" | while read -r line
```

Comment:
Looks a lot better. Might want to add `^` to start at the beginning of a line and/or exclude lines starting with `#`. awk might be perfect for this but happy to keep it like this as well.

Reply:
resolved, see fb0418a

Reply:
To be honest, I have not dived into awk, yet. If it is really beneficial I would go that route, though.